hbase-common/src/main/java/org/apache/hadoop/hbase/util/Strings.java

```diff
@@ -94,4 +102,37 @@ public static String padFront(String input, char padding, int length) {
     int numPaddingCharacters = length - input.length();
     return StringUtils.repeat(padding, numPaddingCharacters) + input;
   }
+
+  /**
+   * Parse the query string of an URI to a key value map. If a single key occurred multiple times,
+   * only the first one will take effect.
+   */
+  public static Map<String, String> parseURIQueries(URI uri) {
+    if (StringUtils.isBlank(uri.getRawQuery())) {
+      return Collections.emptyMap();
+    }
+    return Splitter.on('&').trimResults().splitToStream(uri.getRawQuery()).map(kv -> {
```

Comment:
Why manually decode values when you can call `getQuery()`, which explicitly does the decoding for you?

Reply:
Or do we have `org.apache.http.client.utils.URLEncodedUtils` on the classpath?

Reply:
Not sure but I'm trying to not rely on libs other than guava as much as possible, to minimize our dependency scope...

Reply:
I agree, I would also prioritize a smaller dependency set. There's an [open PR](https://github.com/google/guava/issues/1005) for Guava to add some support, but I guess nothing there yet.